include/envoy/server/resource_monitor_config.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/server/resource_monitor.h"
+
+#include "common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Server {
+namespace Configuration {
+
+/**
+ * Implemented by each resource monitor and registered via Registry::registerFactory()
+ * or the convenience class RegistryFactory.
+ */
+class ResourceMonitorFactory {
+public:
+  virtual ~ResourceMonitorFactory() {}
+
+  /**
+   * Create a particular resource monitor implementation.
+   * @param config const ProtoBuf::Message& supplies the config for the resource monitor
+   *        implementation.
+   * @param dispatcher Event::Dispatcher& the main thread's dispatcher. This is used by
```

Comment:
Optional comment: I would expect that in the future we might need to add other things here beyond the dispatcher, such as cluster manager, TLS, etc. It might be better to just create some type of Factory interface like we do for the other types of plugins? That way it's easier to add additional members to pass in the future without having to change the interface everywhere.